### DISCUSSION TITLE
Next Development server crashes due to module and *.ts file resolution problems

### DISCUSSION THREAD

**Discussion by ghost** [vercel/next.js#72245] [Help] [answered]
### Summary

Next Development server crashes due to module and *.ts file resolution issues

I recently created a project with next.js and its latest version and although initially it worked correctly without any problem with the development server, I was surprised to see that this suddenly changed.

> I use bun as the runtime environment for my projects.

I would like you to tell me how to solve these problems, since I have been looking for information for a while (2-3h) and I don't know where to start, I don't know if it is beyond my reach to solve it or if it is something easier to do. 

Any helpful comments would be greatly appreciated.

Thx

### Additional information

```js
output from my console:


   ▲ Next.js 15.0.2 (Turbopack)
   - Local:        http://localhost:3000
   - Environments: .env.local
   - Experiments (use with caution):
     · turbo

 ✓ Starting...
(node:11304) [DEP0040] DeprecationWarning: The `punycode` module is deprecated. Please use a userland alternative instead.
(Use `node --trace-deprecation ...` to show where the warning was created)
(node:2728) [DEP0040] DeprecationWarning: The `punycode` module is deprecated. Please use a userland alternative instead.
(Use `node --trace-deprecation ...` to show where the warning was created)

 ⚡ Million Lint v1.0.11
 ✓ Ready in 0.63ms
 ⨯ ./node_modules/@mapbox/node-pre-gyp/lib/util/nw-pre-gyp/index.html
Unknown module type
This module doesn't have an associated type. Use a known file extension, or register a loader for it.

Read more: https://nextjs.org/docs/app/api-reference/next-config-js/turbo#webpack-loaders

 ⨯ ./node_modules/fs.realpath
Unknown module type
This module doesn't have an associated type. Use a known file extension, or register a loader for it.

Read more: https://nextjs.org/docs/app/api-reference/next-config-js/turbo#webpack-loaders

 ⨯ ./src/app/lazy-animations.tsx.tsx
Module not found: Can't resolve './lazy-animations.tsx.tsx'

 ⨯ ./node_modules/@mapbox/node-pre-gyp/lib/util/s3_setup.js:76:15
Module not found: Can't resolve 'aws-sdk'
  74 |
  75 |   // if not mocking then setup real s3.
> 76 |   const AWS = require('aws-sdk');
     |               ^^^^^^^^^^^^^^^^^^
  77 |
  78 |   AWS.config.update(config);
  79 |   const s3 = new AWS.S3();



https://nextjs.org/docs/messages/module-not-found

https://nextjs.org/docs/messages/module-not-found


 ⨯ ./node_modules/@mapbox/node-pre-gyp/lib/util/s3_setup.js:43:21
Module not found: Can't resolve 'mock-aws-s3'
  41 |     // here we're mocking. node_pre_gyp_mock_s3 is the scratch directory
  42 |     // for the mock code.
> 43 |     const AWSMock = require('mock-aws-s3');
     |                     ^^^^^^^^^^^^^^^^^^^^^^
  44 |     const os = require('os');
  45 |
  46 |     AWSMock.config.basePath = `${os.tmpdir()}/mock`;
  
   ⨯ ./node_modules/@mapbox/node-pre-gyp/lib/util/s3_setup.js:112:16
Module not found: Can't resolve 'nock'
  110 |   }
  111 |
> 112 |   const nock = require('nock');
      |                ^^^^^^^^^^^^^^^
  113 |   // the bucket used for testing, as addressed by https.
  114 |   const host = 'https://mapbox-node-pre-gyp-public-testing-bucket.s3.us-east-1.amazonaws.com';
  115 |   const mockDir = process.env.node_pre_gyp_mock_s3 + '/mapbox-node-pre-gyp-public-testing-bucket';

 ✓ Compiled in 14.8s
 ✓ Ready in 17.7s
 ⨯ Error [TypeError]: Cannot read properties of undefined (reading '_handle')
    at <unknown> (file://C:\Users\Flaco\Documents\Lost\programming\projects\web\arty\.next\server\edge\chunks\node_modules_5fcb53._.js:18426:20)
    at Array.forEach (<anonymous>)
    at push.94113.module.exports (file://C:\Users\Flaco\Documents\Lost\programming\projects\web\arty\.next\server\edge\chunks\node_modules_5fcb53._.js:18425:7)
    at 56685 (file://C:\Users\Flaco\Documents\Lost\programming\projects\web\arty\.next\server\edge\chunks\node_modules_5fcb53._.js:18444:1)
    at <unknown> (file://C:\Users\Flaco\Documents\Lost\programming\projects\web\arty\.next\server\edge\chunks\edge-wrapper_e40a4b.js:693:27)
    at runModuleExecutionHooks (file://C:\Users\Flaco\Documents\Lost\programming\projects\web\arty\.next\server\edge\chunks\edge-wrapper_e40a4b.js:738:9)
    at instantiateModule (file://C:\Users\Flaco\Documents\Lost\programming\projects\web\arty\.next\server\edge\chunks\edge-wrapper_e40a4b.js:691:9)
    at getOrInstantiateModuleFromParent (file://C:\Users\Flaco\Documents\Lost\programming\projects\web\arty\.next\server\edge\chunks\edge-wrapper_e40a4b.js:624:12)
    at commonJsRequire (file://C:\Users\Flaco\Documents\Lost\programming\projects\web\arty\.next\server\edge\chunks\edge-wrapper_e40a4b.js:157:20)
    at 78746 (file://C:\Users\Flaco\Documents\Lost\programming\projects\web\arty\.next\server\edge\chunks\node_modules_@mapbox_node-pre-gyp_ebb9bb._.js:1880:13)
 ⨯ ./node_modules/@mapbox/node-pre-gyp/lib/util/nw-pre-gyp/index.html
Unknown module type
This module doesn't have an associated type. Use a known file extension, or register a loader for it.

Read more: https://nextjs.org/docs/app/api-reference/next-config-js/turbo#webpack-loaders
```

> Try to summarize the repeated console error messages and this summarizes everything.
```


### Example

_No response_

**Comment by Mahfoudh94** [comment]
This is happening to me right now, everything was sound and clear and then suddenly it all started.
Weirdly enough I am using bun too, but this project I am on uses also Auth.js (formerly NextAuth) and bcrypt, my database is postgres with drizzle orm. do you happen to use any or some of these too?

**Reply by ghost** [reply]
I'm using NextAuth, Bcrypt (I replaced it with bcryptjs since bcrypt was giving me problems), drizzle, frame-motion and also Million-Lint. This last one was the cause of my problems, after removing it everything worked normally again, although it is a great tool it gives me problems when executing the code (which are the ones I showed).

**Comment by Omar-Seba** [comment]
I'm facing the same issue. 
the problem appears when I set up the middleware from nextAuth 
I don't have the error when I create manually a middleware 
I'm using NextAuth v5

**Reply by ghost** [reply]
If you don't have a problem with the middleware, you should check the packages you have installed because one of them must be giving a conflict that prevents the application from running, in my case it was million-lint.

**Comment by heryTz** [comment]
Any solution to this ? I'm facing the same issue

**Reply by ghost** [reply, marked as answer]
This is what worked for me:

- Review the dependencies I have.
- Review the code (middleware, orm, components and others) and make sure there are no conflicts.
- Check the compatibility of some dependencies with the latest version of next-js.
- Look for problems with some dependencies in next-js.
- With this delete some dependencies or modify the code that could be generating the conflict.

Mostly, the problem was because of Million and Million-Lint, it was giving me a conflict when reading my files. I managed to solve the problem but I also forgot a couple of details.

But here is a brief explanation of what Million was doing:

Million is added to your `next.config.ts`.

```ts
import MillionLint from “@million/lint”;
 
const nextConfig = {
  // ...
};
 
export default MillionLint.next({
  rsc: true,
  filter {
     exclude: “**/node_modules/**/*”
     include: “”**/*.{mtsx,mjsx,tsx,jsx}"”
  }
})(nextConfig);
```

This gave me a problem with some files, these were `frame-features.ts` and `lazy-animations.tsx`.

![Files with which I had conflict; frame-features.ts and lazy-animations.tsx](https://github.com/user-attachments/assets/41a54d38-7e4b-4a13-b694-09b4bbd666e2)

And why?

Well...

We hope you will read it as `frame-features.ts` and `lazy-animations.tsx`. But no, I read it as `frame-features.ts.ts` and `lazy-animations.tsx.tsx` (You can see it in the output that throws error at the beginning of the discuction).

For this reason when I started development mode everything crashed.

> next.js suggested me to check the turbo-pack configuration as I didn't understand the file extension and it was a part that also confused me. The problem is not with turbo-pack, but with Million-Lint,

 if you have a similar problem and you haven't found a solution or it doesn't work, don't despair, you will find the solution.
 
 Have some patience ✨

**Comment by jalalDev7** [comment]
Replace bcrypt with bcryptjs in your project and that will  fix the problem

**Reply by bckd00r** [reply]
how?

**Reply by ghost** [reply]
> how?

- Uninstall bcrypt from npm, yarn, pnpm or bun (if you are using typescript also uninstall @types/bcrypt)
- Install bcryptjs (if you are using typescript also install @types/bcryptjs)
- Modify “bcrypt” imports to “bcryptjs”.
- Everything should work
